osquery/tables/system/darwin/gatekeeper.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/sql/sqlite_util.h"
+
+namespace fs = boost::filesystem;
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+const std::string kGkeStatusPath{
+    "/var/db/SystemPolicy-prefs.plist"};
+
+const std::string kGkeBundlePath{
+    "/var/db/gke.bundle/Contents/version.plist"};
+
+const std::string kGkeOpaquePath{
+    "/var/db/gkopaque.bundle/Contents/version.plist"};
+
+const std::string kPolicyDb = "/var/db/SystemPolicy";
+
+bool isGateKeeperDevIdEnabled() {
+  sqlite3* db = nullptr;
+  auto rc = sqlite3_open_v2(
+      kPolicyDb.c_str(),
+      &db,
+      (SQLITE_OPEN_READONLY | SQLITE_OPEN_PRIVATECACHE | SQLITE_OPEN_NOMUTEX),
+      nullptr);
+  if (rc != SQLITE_OK || db == nullptr) {
+    VLOG(1) << "Cannot open Gatekeeper DB: " << rc << " "
+            << getStringForSQLiteReturnCode(rc);
+    if (db != nullptr) {
+      free(db);
+    }
+  }
+
+  std::string query = "SELECT disabled FROM authority WHERE label = 'Developer ID'";
+  sqlite3_stmt* stmt = nullptr;
+  rc = sqlite3_prepare_v2(db, query.c_str(), -1, &stmt, nullptr);
+
+  while ((rc = sqlite3_step(stmt)) == SQLITE_ROW) {
```

Comment:
And this particular `rc` value never gets used, so this can really be simplified to 
```
while (sqlite3_step(stmt) == SQLITE_ROW) {
```